### PR TITLE
Made the name of repo checkout directory depend on the layout base path

### PR DESCRIPTION
**Pull request by odisseus**
This pull request is related to the issue #1370.

### CODE REVIEW

File: src/core/remote.scala [hunk 1]
```diff
@@ -105,24 +104,39 @@ case class Remote(ref: String) {
...
-  def https: String = simplified match {
+  def local(baseDir: Path): Option[String] = {
+    object ExistingDirectory{
+      def unapply(p: String): Option[Path] = {
+        val absolutePath = baseDir.resolve(Path(p)).absolutePath().get
+        if(absolutePath.directory) Some(absolutePath) else None
+      }
+    }
+    ref match {
+      case r"git@.*" => None
```
Comment (propensive) [inline src/core/remote.scala:127]: I'm not sure it's guaranteed to be `git@` for all repositories.

File: src/core/remote.scala [hunk 2]
```diff
@@ -105,24 +104,39 @@ case class Remote(ref: String) {
...
+      case r"https://.*" => None
+      case ExistingDirectory(dir) => Some(str"file://${dir}")
+      case _ => None
+    }
+  }
+
+  def https: Option[String] = simplified match {
```
Comment (propensive) [inline src/core/remote.scala]: I think the `only` extension method could replace `match` and the `wildcard`. There are lots of examples of its usage in `ids.scala`.

**Review by propensive — COMMENTED** [review]
This can be merged when you think it's ready, and the tests are passing. Looks like there's only one simple line missing from one of the test checks.

File: src/core/remote.scala (hunk 2 above)
Comment (odisseus) [reply src/core/remote.scala]: Refactored

**Review by propensive — APPROVED** [review]

File: src/core/remote.scala (hunk 1 above)
Comment (odisseus) [reply src/core/remote.scala:127]: You are right, Git [supports](https://www.git-scm.com/docs/git-clone#_git_urls) many different address formats and transport protocols. But the assumption that a "git@" prefix means SSH and vice versa already existed in the code before this pull request.

**Pull request merged**